onnx/defs/nn/old.cc

```diff
@@ -1261,20 +1261,7 @@ void convTransposeShapeInference1(InferenceContext& ctx) {
     if ((nullptr != auto_pad_attr) && (auto_pad_attr->s() != "VALID")) {
       int input_dims_size = static_cast<int>(n_input_dims);
       for (int i = 0; i < input_dims_size; ++i) {
-        int64_t residual = 0;
-        int64_t stride = strides[i];
-        if (stride > 1) {
-          if (!input_shape.dim(2 + i).has_dim_value()) {
-            continue;
-          }
-          residual = input_shape.dim(2 + i).dim_value();
-          while (residual >= stride) {
-            residual -= stride;
-          }
-        }
-        int64_t total_pad = residual == 0
-            ? effective_kernel_shape[i] - stride
-            : effective_kernel_shape[i] - residual;
+        int64_t total_pad = effective_kernel_shape[i] - strides[i];
```

Comment:
Wondering what happens with the counterexample you give and onnxruntime? Thanks!

Reply:
FWIW: it looks like this is calculated here: https://github.com/microsoft/onnxruntime/blob/97659495d982982a06eee6a4e3560923b988ff7e/onnxruntime/core/providers/cpu/nn/conv_transpose_attributes.h#L174

Reply:
The example I gave in the PR description is fixed by this. What I did here was just to bring [this change](https://github.com/onnx/onnx/pull/3188/files#diff-b9e98b88715e8a7e56c0dfa317f21a39582f919c87185f827c4a011ad9010639L1228) to the shape inference of the old ConvTranspose with opset < 11. There is no functional difference between opset < 11 and opset >=11 in ConvTranspose, as opset 11 only introduced clarifications in the documentation. So they should behave the same way, which is how the "new" operator behaves.

Reply:
Thanks, sounds good to me.